app/models/manageiq/providers/azure_stack/inventory/parser/cloud_manager.rb

```diff
@@ -71,12 +72,23 @@ def vms
         :vm_or_template  => vm_obj,
         :cpu_sockets     => persister.flavors.lazy_find(flavor_ref, :key => :cpu_cores),
         :cpu_total_cores => persister.flavors.lazy_find(flavor_ref, :key => :cpus),
-        :memory_mb       => persister.flavors.find(flavor_ref).memory,
+        :memory_mb       => persister.flavors.find(flavor_ref).memory / 1.megabytes,
```

Comment:
Hm didn't see the `.find` before but this should really be a `lazy_find` otherwise we can run into order issues.
I'm okay with this for now because there is no targeted refresh here yet and lazy_find doesn't support being given a lambda yet.

Can you open an issue on https://github.com/ManageIQ/inventory_refresh/issues to add support for a lambda to https://github.com/ManageIQ/inventory_refresh/blob/master/lib/inventory_refresh/inventory_object_lazy.rb ?